### PR TITLE
feat: improve backend take-home test

### PR DESCRIPTION
**Pull request by aporcupine**
## :spiral_notepad: What's being changed?

* Upgrade to Go 1.22
* Switch from RabbitMQ to Pubsub (with the emulator)
* Update the API example code to use slog + the stdlib http mux

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,56 +1,58 @@
 # go-record-label-task 🎧 💿
 
-## Dependencies
+## Example Dependencies
```
Comment (aporcupine) [inline README.md:3]: Chat GPT helped with this re-write, I think I got rid of the GPTisms 😂

File: README.md [hunk 2]
```diff
@@ -1,56 +1,58 @@
...
+  incorporate into your solution.
+  - Additional information on the PubSub emulator can be found here: [PubSub Emulator Documentation](https://cloud.google.com/pubsub/docs/emulator)
 
-* Your solution should prioritise maintainability as well as robustness, and utilise a queue/message broker for communications.
+- Your solution should prioritize maintainability and robustness, utilizing a queue/message broker for communications.
 
-* You must provide your complete solution as a zip file, including the `.git` subfolder, and return it privately via email
-(please do not publish your solution publicly!)
-    * Please use your commits to demonstrate incremental progress, as you would whilst working on tickets as part of a sprint.
+- Provide your complete solution as a zip file, and include the output of git log to show your progress along the way.
```
Comment (ryantremarco) [inline README.md:41]: Why make them send us the `git log` output if we have the full zip and can check it ourselves?

File: services/api/domain/types.go [hunk 3]
```diff
@@ -0,0 +1,16 @@
+package domain
+
+import "time"
+
+type ReleaseDistribution struct {
+	Type string `json:"type"`
+	Qty  int64  `json:"qty"`
```
Comment (ryantremarco) [inline services/api/domain/types.go:7]: Can we use `Quantity` and `quantity` just for style consistency with the rest?

**Review by AdamJSc — DISMISSED** [review]
Small comment otherwise all good 👍

File: README.md (hunk 2 above)
Comment (aporcupine) [reply README.md:41]: I like reviewing the code as a PR in the submissions repo, and while possible, trying to keep the git commit history, and raise it as a PR in that repo is a pain in the arse. So figured that git log would do.

File: docker-compose.yml [hunk 4]
```diff
@@ -1,18 +1,28 @@
-version: '3.8'
-
 services:
 
   api:
-    image: golang:1.17
+    image: golang:latest
```
[the remark this answers is not in the document]
Comment (aporcupine) [reply docker-compose.yml:4]: In my opinion we should always be building with the latest version of Go, including in our tests. The go 1 compatibility promise should make this perfectly safe, but happy to discuss that further.

I just looked and it looks like they do have `1` tag, if we wanted to make sure things didn't break if they ever decide to go for a version 2.

[the remark this answers is not in the document]
Comment (AdamJSc) [reply docker-compose.yml:4]: No worries, happy to leave as it is for now then 👍

**Review by ryantremarco — APPROVED** [review]

**Review by JWBfx — APPROVED** [review]

**Pull request merged**